lib/grape/api.rb

```diff
@@ -141,7 +156,7 @@ def instance_for_rack
       # Adds a new stage to the set up require to get a Grape::API up and running
       def add_setup(method, *args, &block)
         setup_step = { method: method, args: args, block: block }
-        @setup << setup_step
+        @setup << setup_step unless @finalized
```

Comment:
When does this happen? Should this be an exception?

Add `finalized?` and `unless finalized?`.

Reply:
@dblock Which exception do you mean? Here we stop adding method calls to the @setup, for example:

```ruby
Api.get '/' do
  # logic here
end
```

It won't be added after calling `finalize!`